File: Data Structures & Algorithms/maximum-product-subarray/submission-1.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        res=max(nums)

        curMin,curMax=1,1

        for n in nums:
            if n==0:
                curMin,curMax=1,1
                continue
            temp=curMax*n
            curMax=max(curMax*n, curMin*n,n)
            curMin=min(temp,curMin*n,n)
            res=max(res,curMax)
        return res

File: Data Structures & Algorithms/duplicate-integer/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums) {
        let set= new Map();
        for(let i=0;i<nums.length;i++){
            if(set.get(nums[i])){
                return true;
            }
            else{
                set.set(nums[i],i+1);
            }
        }
        return false;
    }
}
